server/routes/users.js

```diff
@@ -2,14 +2,272 @@
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all companies a user is interested in 
+router.get('/:user_id/companies', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT company_id, rank FROM user_company_rank WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//Get all events a user is attending
+router.get('/:user_id/events', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM event_checkin WHERE user_id = $1 UNION SELECT event_id FROM event_registration WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all events a user is hosting
+router.get('/:user_id/host', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM event_host WHERE host_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all the user's majors
+router.get('/:user_id/majors', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT major_id FROM user_major WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all positions a user is seeking
+router.get('/:user_id/positions', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT position_id FROM user_position WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all the user's occupations
+router.get('/:user_id/occupations', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT occupation_id FROM user_occupation WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//Get the user's diet information
+router.get('/:user_id/diet', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT diet_id FROM user_diet WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get a user's favorite events
+router.get('/:user_id/favorite', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM favorite_events WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get a user's favorite events
+//HELP NEEDED
+/*
+router.get('/search', function(req, res, next) {
```

Comment:
@nikhitashanker use fix discussed in person. 

Note: I see you have this commented code and the route below as well. They have the same base route and should be the same function with checks to see whether the query parameter is set/valid or not.